cpu_features.h

```diff
@@ -54,7 +54,9 @@ extern uint32_t adler32_power8(uint32_t adler, const unsigned char* buf, size_t
 #ifdef X86_PCLMULQDQ_CRC
 extern uint32_t crc32_fold_reset_pclmulqdq(crc32_fold *crc);
 extern void     crc32_fold_copy_pclmulqdq(crc32_fold *crc, uint8_t *dst, const uint8_t *src, size_t len);
+extern void     crc32_fold_pclmulqdq(crc32_fold *crc, const uint8_t *src, size_t len, uint32_t init_crc);
```

Comment:
Do we need this `crc32_fold_pclmulqdq` function to be extern?

Reply:
I"m pretty sure we do, it's assigned into functable and we need some assurance that that function is declared somewhere.  We mostly want to prevent the issue of it being multiply declared, as that's a warning that newer GCCs actually treat as an error.  Though, this mostly applies to variable declarations and not so much function prototypes.